.github/workflows/release-azure-webapp-v1.yaml

```diff
@@ -0,0 +1,268 @@
+on:
+  workflow_call:
+    inputs:
+      workspace_name:
+        description: The name of the workspace to create the artifact for.
+        type: string
+        required: true
+      resource_group_name:
+        description: Web App resource group name.
+        type: string
+        required: true
+      web_app_name:
+        description: Web App name.
+        type: string
+        required: true
+      environment:
+        description: Environment where the artifact will be deployed.
+        type: string
+        required: true
+      use_staging_slot:
+        description: Use a staging slot for the deployment.
+        type: boolean
+        required: false
+        default: true
+      disable_auto_staging_deploy:
+        description: Disable automatic deployments to the staging slot.
+        type: boolean
+        required: false
+        default: false
+      use_private_agent:
+        description: Use a private agent to deploy the built artifact.
+        type: boolean
+        required: false
+        default: true
+      use_labels:
+        description: Use labels to start the right environment's GitHub runner. If use_labels is true, also use_private_agent must be set to true
```

Comment:
I'd try to be more clear than "the right environment's GitHub runner" (in which sense "the right" ?)

Reply:
@gunzip I don't know either and I don't like it, but I decided that changing that is out of scope. `use_labels` and `override_labels` are present in almost all dx workflows. We need to simplify that, also because the `runs-on` property on jobs is unreadable.